src/app/movie-details/movie-details.component.ts

```diff
@@ -1,40 +1,82 @@
+import { MoviesGenres, Movies, Credits, CreditsResult, AddMovie, MoviesService, Token } from './../movies.service';
 import { Component, OnInit } from '@angular/core';
-import { Movies, MoviesGenres } from '../movies.service';
 import { ActivatedRoute } from '@angular/router';
 import { HttpClient } from '@angular/common/http';
-
+import { environment } from 'src/environments/environment';
+import { Observable } from 'rxjs/internal/Observable';
 
 @Component({
   selector: 'app-movie-details',
   templateUrl: './movie-details.component.html',
   styleUrls: ['./movie-details.component.css']
 })
+
 export class MovieDetailsComponent implements OnInit {
-  imagePath = "https://image.tmdb.org/t/p/w400"
-  public id: any
-  public data: any 
-  genres: Array<MoviesGenres> | undefined;
+  urlImage = environment.urlImage
+  private _id: any
+  persons: Array<CreditsResult> | undefined
+  data$!: Observable<Movies>
+  genres: Array<MoviesGenres> | undefined
+  persons$: Observable<Credits> | undefined
+  permission = environment.authenticate
+  token$: Observable<Token> | undefined
+  request_token: string | undefined
+  name: string | undefined
+
   constructor(
     private route: ActivatedRoute,
-    private http: HttpClient
+    private http: HttpClient,
+    private moviesService: MoviesService,
   ) { }
   
-  
   ngOnInit(): void {
-    this.id = this.route.snapshot.params["id"] 
+    this._id = this.route.snapshot.params["id"] 
+    this.data$ = this.http.get<Movies>(`${environment.apiUrl}/movie/${this._id}${environment.apiKey}`);
+    this.persons$ = this.http.get<Credits>(`${environment.apiUrl}/movie/${this._id}/credits${environment.apiKey}`);
   }
-  getTitles(){
-    const url = "https://api.themoviedb.org/3/movie/"+this.id+"?api_key=38193385b589296926c46f16b67e1b93"
-    this.http.get<Movies>(url).subscribe((res)=>{
-      this.data = res
-      this.genres = res.genres.map(array => this.convertToMovies(array))
-    })
+
+  addMovie (){
+    const movie : AddMovie = ({
+      media_type: "movie",
+      media_id: this._id,
+      watchlist: true
+    });
+    this.moviesService.sendMovie(movie).subscribe(id => {
+      console.log(id);
+  })} 
+
+  getToken(){
+  this.moviesService.getToken().subscribe(request_token =>
+    this.request_token = request_token.request_token)
   }
- 
-  convertToMovies (dto: any) : MoviesGenres {
-    return { 
-      name: dto.name
-    }
+
+  test(dane: string){
+    this.request_token = dane
   }
+  
+  sendToken(data: string){
+   this.moviesService.sendRequestTokenn(data)
+  }
+
+  watchlist(){
+    if(this.moviesService.sessionId != undefined){
+      this.addMovie ()
+      alert("added to watchlist")
+    }
+    else if(this.request_token != undefined){
```

Comment:
Logika z tej funkcji powinna być w moviesService.
Dodatkowo funkcje powinny sendToken i addMovie być robione po kolei, czyli jak skończy się sendToken to wywołać addMovie